src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/HttpHeadersReader.cs

```diff
@@ -36,7 +43,25 @@ public void ReadRequestHeaders(HttpRequestMessage request, List<KeyValuePair<str
             return;
         }
 
-        ReadHeaders(request.Headers, _requestHeaders, destination);
+        ReadHeaders(request.Headers, _requestHeadersToLog, destination);
+        if (request.Content is not null)
```

Comment:
Since this is a change in behavior that could potentially result in a lot of extra telemetry and perhaps a privacy incident, this feature probably needs to be controlled by an option.

Reply:
Good point, done